lib/blocs/chat_bloc.dart

```diff
@@ -358,12 +371,14 @@ class ChatBloc {
   }
 }
 
-extension Archived on RxList<Chat> {
+extension Helpers on RxList<Chat> {
   /// Helper to return archived chats or all chats depending on the bool passed to it
   /// This helps reduce a vast amount of code in build methods so the widgets can
   /// update without StreamBuilders
   archivedHelper(bool archived) {
-    if (archived) return this.where((e) => e.isArchived ?? false).toList();
-    else return this;
+    if (archived)
+      return this.where((e) => e.isArchived ?? false).toList();
+    else
+      return this.where((e) => !(e.isArchived ?? false)).toList();
```

Comment:
I think the goal here was to return *all* chats when `archived` was false, but it seems you updated it to return non-archived chats? I think we should do it this way but I'm kind of curious why it was working correctly before lol.

Reply:
The reason it was working before was that we had a check in the main chat list to return a `Container()` if we were in archive mode and the chat wasn't archived or vice versa. This fix eliminated the need for that extra code as well.

This was the code (that I got rid of)
`if (!showArchived && ChatBloc().chats.archivedHelper(showArchived)[index].isArchived!) return Container();
if (showArchived && !ChatBloc().chats.archivedHelper(showArchived)[index].isArchived!) return Container();`

Reply:
Ahhh good looks thanks!